CMakeLists.txt

```diff
@@ -842,6 +843,28 @@ add_custom_target(${DRIVER_VALIDATION_H_TARGET} DEPENDS ${DRV_VALIDATION})
 include(${ZEPHYR_BASE}/cmake/kobj.cmake)
 gen_kobj(KOBJ_INCLUDE_PATH)
 
+# Generate sections for kernel device subsystems
+set(
+  DEVICE_API_LD_SECTIONS
+  ${CMAKE_CURRENT_BINARY_DIR}/include/generated/device-api-sections.ld
+  )
+
+add_custom_command(
+  OUTPUT ${DEVICE_API_LD_SECTIONS}
+  COMMAND
+    ${PYTHON_EXECUTABLE}
+    ${ZEPHYR_BASE}/scripts/build/gen_iter_sections.py
+    -i ${struct_tags_json}
+    -t __subsystem
+    -o ${DEVICE_API_LD_SECTIONS}
+  DEPENDS
+    ${ZEPHYR_BASE}/scripts/build/gen_iter_sections.py
+    ${struct_tags_json}
```

Comment:
It's a bit confusing to me that I don't find the producer of this target file in your PR (or am I really so blind?). Guess this should be parse_syscalls? Maybe the answer is somewhere hidden inside the 100+ comments, but a search for "json" didn't produce an answer...

Reply:
Solved. It's an existing file therefore not in the PR.